sympy/solvers/decompogen.py

```diff
@@ -32,7 +32,8 @@ def decompogen(f, symbol):
     """
     f = sympify(f)
     result = []
-
+    if isinstance(f, Relational):
```

Comment:
How about `if not isinstance(f, Relational): raise TypeError`? Before making and committing such a change, however, it would be good to make the change and run the decompogen test suite locally.